apple/ios.bzl

```diff
@@ -79,6 +80,37 @@ def ios_app_clip(name, **kwargs):
         **bundling_args
     )
 
+def ios_dylib(name, **kwargs):
+    # buildifier: disable=function-docstring-args
+    """Builds a iOS dylib."""
+
+    # Xcode will happily apply entitlements during code signing for a dylib even
+    # though it doesn't have a Capabilities tab in the project settings.
+    # Until there's official support for it, we'll fail if we see those attributes
+    # (which are added to the rule because of the code_signing_attributes usage in
+    # the rule definition).
+    if "entitlements" in kwargs or "provisioning_profile" in kwargs:
+        fail("ios_dylib does not support entitlements or provisioning " +
+             "profiles at this time")
+
+    binary_args = dict(kwargs)
+
+    dylib_args = binary_support.add_entitlements_and_swift_linkopts(
+        name,
+        platform_type = str(apple_common.platform_type.ios),
+        product_type = apple_product_type.dylib,
+        link_swift_statically = True,
+        include_entitlements = False,
+        exported_symbols_lists = binary_args.pop("exported_symbols_lists", None),
+        **binary_args
+    )
+
+    _ios_dylib(
+        name = name,
+        dylibs = binary_args.pop("frameworks", None),
```

Comment:
why would this be passed in via `frameworks`?

Reply:
This allows specifying `ios_framework` deps, the same as `ios_application`.

Reply:
like the dylib can have frameworks?

Reply:
It can depend on them, yes.